src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1328,10 +1330,15 @@ protected function castAttributeAsJson($key, $value)
      * Encode the given value as JSON.
      *
      * @param  mixed  $value
+     * @param  bool  $isUnicode
      * @return string
      */
-    protected function asJson($value)
+    protected function asJson($value, $isUnicode = false)
```

Comment:
I don't like idea of adding a single-use parameter to `asJson`. What if we wanted to add (for whatever reason), the ability to do `JSON_PRETTY_PRINT`? Or maybe `JSON_PARTIAL_OUTPUT_ON_ERROR`?

I think it should be `asJson($value, int $flags = 0)` and we can add an additional method to process the flags.

Maybe something like this?
```php
protected function getJsonCastFlags($key): int
{
    $flags = 0;

    if ($this->hasCast($key, ['json:unicode'])) {
        $flags |= JSON_UNESCAPED_UNICODE;
    }

    return $flags;
}
```

Reply:
Thank you for your review! 
I've addressed your feedback in commit [61fad07](https://github.com/laravel/framework/pull/54992/commits/61fad079c593a50e2f8069366ee785246a3ac1ca). 
Please let me know if there's anything else that needs adjustment.